injective_functions/auction/__init__.py

```diff
@@ -13,10 +16,35 @@ def __init__(self, chain_client) -> None:
         super().__init__(chain_client)
 
     async def send_bid_auction(self, round: int, amount: str) -> Dict:
-
+        await self.chain_client.init_client()
         msg = self.chain_client.composer.MsgBid(
             sender=self.chain_client.address.to_acc_bech32(),
             round=round,
             bid_amount=Decimal(amount),
         )
         return await self.chain_client.build_and_broadcast_tx(msg)
+
+    async def fetch_auctions(self) -> Dict:
+        try:
+            auctions = await self.chain_client.client.fetch_auctions()
+            return {"success": True, "result": auctions["auctions"]}
+        except Exception as e:
+            return {"success": False, "error": detailed_exception_info(e)}
+
+    async def fetch_latest_auction(self) -> Dict:
```

Comment:
I'd add a little more safety to this.. something like.. (and ensuring that we return an error on the Exception case)
  ```
try:
        result = await self.fetch_auctions()
        
        # Ensure the result structure is as expected
        if result.get("success") and result.get("result"):
            latest_auction = result["result"][-1]  # Safely access the latest auction
            return {"success": True, "result": latest_auction}
        
except Exception as e:
        return {"success": False, "error": f"Unexpected error: {str(e)}"}
```

Reply:
uses standardized exceptions to give us detailed info on errors, so all the functions either return {"success": True, "result": Dict} or {"sucess": False, "result": Dict} this ensures the gpt function calling easily reads through errors